tests/5.0/target_data/test_target_data_use_device_addr.F90

```diff
@@ -0,0 +1,67 @@
+!===--- test_target_data_use_device_addr.F90 ------------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This file is a test for the use_device_addr when used with the map clause with
+! target data directive. This test uses a scalar and an array of size N which values
+! are modified on the  device and tested in the host. List items that  appear in a
+! use_device_addr clause have the address of the corresponding object in the device
+! data environment inside the construct. This test also tests that address conversions
+! of use_device_addr clauses will occur as if performed after all variables are mapped
+! according to those map clauses.
+!
+!===-----------------------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+PROGRAM test_target_data_use_device_addr
+   USE iso_fortran_env
+   USE ompvv_lib
+   USE omp_lib
+   implicit none
+
+   REAL, POINTER :: dev_ptr
+   INTEGER :: errors, host_data, device_data
+
+   errors = 0
+   device_data = 14
+   host_data = 0 
+
+   ALLOCATE(dev_ptr)
+
+   OMPVV_TEST_OFFLOADING
+
+   OMPVV_TEST_VERBOSE(test(dev_ptr) .ne. 0)
+
+   OMPVV_REPORT_AND_RETURN()
+
+
+CONTAINS
+   INTEGER FUNCTION test(x)
+      REAL, INTENT(INOUT) :: x
+
+      !$omp target data map(to: device_data)
+
+         !$omp target data use_device_addr(device_data)
+            x = device_data
```

Comment:
I do not understand that line:  `x` is a variable on the host; `device_data` is a references to the "corresponding storage in the device data environment ."

But as we are on the host, `device_data`'s device data cannot be accessed. (Exception: unified-shared memory.)

I think one could do: `dev_ptr = c_loc(device_data)`  to obtain the device address but not much more.